docs/ref/csp.txt

```diff
@@ -0,0 +1,296 @@
+.. _content-security-policy:
+
+=============================
+Content Security Policy (CSP)
+=============================
+
+.. versionadded:: 5.2
+
+.. module:: django.middleware.csp
+   :synopsis: Middleware for Content Security Policy headers
+
+Content Security Policy (CSP) is a web security standard that helps prevent
+attacks by controlling the sources from which content can be loaded. CSP
+enforcement plays an important role in a comprehensive :doc:`security strategy
+</topics/security>`.
+
+.. _using-csp:
+
+Using CSP
+=========
+
+To enable CSP in your Django project:
+
+1. Add the CSP middleware to your ``MIDDLEWARE`` setting::
+
+      MIDDLEWARE = [
+          # ...
+          "django.middleware.csp.ContentSecurityPolicyMiddleware",
+          # ...
+      ]
+
+2. Configure your CSP policies in your ``settings.py`` using either
+   ``SECURE_CSP`` or ``SECURE_CSP_REPORT_ONLY``::
```

Comment:
What happens if I set both? Should we be saying `and` here, one for each header, both of which can be used at the same time?

Reply:
Setting both is good! Using the report-only CSP as the test bed for new policies is a good idea. The wording should be updated to suggest that - it wasn't intended to be one or the other, more that you configure either one as needed.